File: SortVision/public/code/bubble/dart/bubbleSort.dart
class BubbleSort {
  // Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
  static void sort(List<int> arr) {
    for (int i = 0; i < arr.length - 1; i++) {
      bool swapped = false;
      for (int j = 0; j < arr.length - i - 1; j++) {
        if (arr[j] > arr[j + 1]) {
          final t = arr[j];
          arr[j] = arr[j + 1];
          arr[j + 1] = t;
          swapped = true;
        }
      }
      if (!swapped) break;
    }
  }
}

void main() {
  final arr = [64, 34, 25, 12, 22, 11, 90];
  BubbleSort.sort(arr);
  print(arr);
}

File: SortVision/public/code/bubble/php/bubbleSort.php
<?php
// Bubble Sort (in-place)
// Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
function bubbleSort(array &$arr): void {
    $n = count($arr);
    for ($i = 0; $i < $n - 1; $i++) {
        $swapped = false;
        for ($j = 0; $j < $n - $i - 1; $j++) {
            if ($arr[$j] > $arr[$j + 1]) {
                [$arr[$j], $arr[$j + 1]] = [$arr[$j + 1], $arr[$j]];
                $swapped = true;
            }
        }
        if (!$swapped) break;
    }
}

$arr = [64, 34, 25, 12, 22, 11, 90];
bubbleSort($arr);
echo implode(', ', $arr) . PHP_EOL;
?>
